sigar_openbsd.go

```diff
@@ -315,6 +315,45 @@ func (self *Cpu) Get() error {
 }
 
 func (self *CpuList) Get() error {
+	mib := [2]int32{C.CTL_HW, C.HW_NCPU}
+	var ncpu int
+
+	n := uintptr(0)
+	// First we determine how much memory we'll need to pass later on (via `n`)
+	_, _, errno := syscall.Syscall6(syscall.SYS___SYSCTL, uintptr(unsafe.Pointer(&mib[0])), 2, 0, uintptr(unsafe.Pointer(&n)), 0, 0)
+
+	if errno != 0 || n == 0 {
+		return nil
+	}
+
+	// Now perform the actual sysctl(3) call, storing the result in ncpu
+	_, _, errno = syscall.Syscall6(syscall.SYS___SYSCTL, uintptr(unsafe.Pointer(&mib[0])), 2, uintptr(unsafe.Pointer(&ncpu)), uintptr(unsafe.Pointer(&n)), 0, 0)
+
+	if errno != 0 || n == 0 {
+		return nil
+	}
+
+	load := [C.CPUSTATES]C.long{C.CP_USER, C.CP_NICE, C.CP_SYS, C.CP_INTR, C.CP_IDLE}
+
+	self.List = make([]Cpu, 0, ncpu)
+
+	// Slightly different semantics for single CPU vs. multiple CPU
+	if ncpu == 1 {
```

Comment:
It looks like you could change the code to not require separate handling of `ncpu == 1`. I think this would work:

```
self.List = make([]Cpu, ncpu)
for _, cpu := range self.List {
    sysctlCptime(ncpu, 1, &load)
    fillCpu(&cpu, load)
}